packages/shared/src/contexts/AuthContext.tsx

```diff
@@ -121,6 +124,11 @@ export const AuthContextProvider = ({
   const [loginState, setLoginState] = useState<LoginState | null>(null);
   const endUser = user && 'providers' in user ? user : null;
   const referral = user?.referrer;
+  const { data: actions } = useQuery(
+    generateQueryKey(RequestKey.Actions, endUser),
+    getUserActions,
+    { enabled: !!endUser?.id },
+  );
```

Comment:
I think it is better to just move this into the `useActions` hook. I don't now if there are any benefits of tying this to `AuthContext` (we can always wait for user id there), we would also expose `isLoading` and other things for actions easier if query us just inside the hook. Also if for any reasons `actions` become heavy it is better to have it outside of AuthContext where we can then fetch only when needed.

Reply:
My concern is that whenever we initialize the `useActions` it will fetch the actions data. We could have instantiated on other places where we just need the completion function. So when a refetch triggers (let's say refocus) it could be triggered on 5 different areas which are unnecessary.

> Also if for any reasons actions become heavy it is better to have it outside of AuthContext where we can then fetch only when needed.

The call for the `actions` is async so it won't be affecting any other areas. Also that we discussed to fetch this only once on load after boot.

Reply:
I agree with Ante, don't think this should be part of the auth context.
Even rather just remove it for now and tackle in separate PR maybe?

Reply:
> I agree with Ante, don't think this should be part of the auth context.

Alright, let's keep the query inside the hook.

> Even rather just remove it for now and tackle in separate PR maybe?

The hook itself is needed here, so decided to create it now than be blocked.